.github/workflows/build-release-binaries.yml

```diff
@@ -0,0 +1,104 @@
+name: Build and Release Binaries
+
+on:
+  workflow_run:
+    workflows: ["Publish Release"]
+    types:
+      - completed
+
+jobs:
+  build-binaries:
+    # Only run if the triggering workflow succeeded and it was triggered by a tag
+    if: >
+      github.event.workflow_run.conclusion == 'success' &&
+      startsWith(github.event.workflow_run.head_branch, 'refs/tags/v')
+    runs-on: ubuntu-24.04
+    name: Build Cross-Platform Binaries
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.23.4'
```

Comment:
Can we reference https://github.com/ObolNetwork/charon/blob/main/.github/actions/setup-go/action.yml to we remain consistent on golang version used across all Actions?